### ISSUE TITLE
Inconsistent Behavior noted when input box has nothing in it

### ISSUE THREAD

**Issue by sajidbarcha** [vanshg395/intl_phone_field#153]
I have noticed that if the input box has not been touched (i.e. user has not input value yet), the _formKey.currentState!.validate() returns TRUE, but if a user inputs a value and then deletes everything, then _formKey.currentState!.validate() returns false. 

In both case there is no value in the input box IntlPhoneField's controller's text, but the latter case makes entering a phone number mandatory to proceed, whereas if the user ignores the IntlPhoneField altogether, they can proceed as if this field is optional. 

I think this we should be able to accept "" values in this field to make it optional, or even add an attribute on IntlPhoneField called isOptional (bool) which validates the IntlPhoneField as valid even if a user inputs something and then deletes it.

**Comment by hamed-ta** [comment]
I noticed the same problem, another thing is that if the user doesn't touch the field the validator doesn't check the entry so if the form is empty it will validate the form, and another thing is that if I set AutoValidateMode to disable the field's validator won't get called at all. and what if I want to only validate the field when I call form _formKey.currentState!.validate()?

**Comment by philitell** [comment]
Yes, there is definitly an inconsistent behaviour:
If you don't touch the phone-number field and the form-state is saved via `_formKey.currentState!.save();` then the onSave-method of the field returns null.
If you enter a phone number, delete it immediatly (form field is empty again and has no focus) and save the form-state then the onSave-method returns a phonenumber which is only the country code (e.g. "+43") - whereas null would be expected in this case.

long story short: value of phoneNumber should always be null if textfield value is ""

```
return PhoneFormField(
      controller: _phoneNumberController,
      shouldFormat: false,
      defaultCountry: 'AT',
      decoration: const InputDecoration(
        labelText: "Telefonnummer",
        border: OutlineInputBorder(),
      ),
      onSaved: (phoneNumber) {...}

    );
```
